src/include/packing_inline.h

```diff
@@ -754,3 +754,294 @@ __wt_struct_size_adjust(WT_SESSION_IMPL *session, size_t *sizep)
+    __pack_decode__10(p, maxlen, T5, V5, T6, V6, T7, V7, T8, V8, T9, V9);
+#define __pack_decode__20(p, maxlen, T1, V1, T2, V2, T3, V3, T4, V4, T5, V5, T6, V6, T7, V7, T8, V8, T9, V9, T10, V10)  \
+    __pack_decode__10(p, maxlen, T1, V1, T2, V2, T3, V3, T4, V4, T5, V5); \
+    __pack_decode__10(p, maxlen, T6, V6, T7, V7, T8, V8, T9, V9, T10, V10);
+
+#define __pack_decode_direct(p, maxlen, ...)   do { WT_CONCAT(__pack_decode__, WT_NARG(__VA_ARGS__))(p, maxlen, __VA_ARGS__) } while (0)
+
+
+/*
+ * Macros for defining implementations of packing functions
+ */
+
+#define WT_NARG_FUNCARGS_2(T1, V1)  T1 *V1
+#define WT_NARG_FUNCARGS_4(T1, V1, T2, V2)  T1 *V1, T2 *V2
+#define WT_NARG_FUNCARGS_6(T1, V1, T2, V2, T3, V3)  T1 *V1, T2 *V2, T3 *V3
+#define WT_NARG_FUNCARGS_8(T1, V1, T2, V2, T3, V3, T4, V4)  T1 *V1, T2 *V2, T3 *V3, T4 *V4
+#define WT_NARG_FUNCARGS_10(T1, V1, T2, V2, T3, V3, T4, V4, T5, V5)  T1 *V1, T2 *V2, T3 *V3, T4 *V4, T5 *V5
+#define WT_NARG_FUNCARGS_12(T1, V1, T2, V2, T3, V3, T4, V4, T5, V5, T6, V6)  T1 *V1, T2 *V2, T3 *V3, T4 *V4, T5 *V5, T6 *V6
+#define WT_NARG_FUNCARGS_14(T1, V1, T2, V2, T3, V3, T4, V4, T5, V5, T6, V6, T7, V7)  T1 *V1, T2 *V2, T3 *V3, T4 *V4, T5 *V5, T6 *V6, T7 *V7
+#define WT_NARG_FUNCARGS_16(T1, V1, T2, V2, T3, V3, T4, V4, T5, V5, T6, V6, T7, V7, T8, V8)  T1 *V1, T2 *V2, T3 *V3, T4 *V4, T5 *V5, T6 *V6, T7 *V7, T8 *V8
+#define WT_NARG_FUNCARGS_18(T1, V1, T2, V2, T3, V3, T4, V4, T5, V5, T6, V6, T7, V7, T8, V8, T9, V9)  T1 *V1, T2 *V2, T3 *V3, T4 *V4, T5 *V5, T6 *V6, T7 *V7, T8 *V8, T9 *V9
+#define WT_NARG_FUNCARGS_20(T1, V1, T2, V2, T3, V3, T4, V4, T5, V5, T6, V6, T7, V7, T8, V8, T9, V9, T10, V10)  T1 *V1, T2 *V2, T3 *V3, T4 *V4, T5 *V5, T6 *V6, T7 *V7, T8 *V8, T9 *V9, T10 *V10
+
+#define WT_NARG_FUNCARGS(...)   WT_CONCAT(WT_NARG_FUNCARGS_, WT_NARG(__VA_ARGS__))(__VA_ARGS__)
+
+
+#define WT_DEFINE_PACKING(NAME, ...) \
+    static inline int \
+    __wt_size_##NAME(WT_SESSION_IMPL *session, size_t *sizep, WT_NARG_FUNCARGS(__VA_ARGS__)) \
+    { \
+        WT_UNUSED(session); \
+        *sizep = __pack_size_direct(__VA_ARGS__); \
+        return (0); \
+    } \
+    static inline int \
+    __wt_pack_##NAME(WT_SESSION_IMPL *session, uint8_t *p, uint8_t *end, WT_NARG_FUNCARGS(__VA_ARGS__)) \
+    { \
+        WT_UNUSED(session); \
+        __pack_encode_direct(p, end, __VA_ARGS__); \
+        return (0); \
+    } \
+    static inline int \
+    __wt_unpack_##NAME(WT_SESSION_IMPL *session, const uint8_t **pp, size_t size, WT_NARG_FUNCARGS(__VA_ARGS__)) \
+    { \
+        WT_UNUSED(session); \
+        __pack_decode_direct(pp, size, __VA_ARGS__); \
+        return (0); \
+    }
+
+
+/*
+ * Specialized packing functions implementations
+ */
+
+WT_DEFINE_PACKING(system_record,  /* __wt_size_system_record(), __wt_pack_system_record(), __wt_unpack_system_record() */
+    uint32_t, rectype)
+
+WT_DEFINE_PACKING(checkpoint_start,  /* __wt_size_checkpoint_start(), __wt_pack_checkpoint_start(), __wt_unpack_checkpoint_start() */
```

Comment:
I haven't dug deeply into what this replaces, but I liked that the prior version just required a change to "dist/log_data.py" (which allows for records to be described in a natural way), then the relevant code was generated and correct.

It is nice not having to manually edit multiple locations when adding log record types, especially when it is done so rarely.

Reply:
Yeah I have to agree. Variadic macros don't work with older MSVC and it doesn't fit into multi-inclusion pattern.

I'll revert back to the Python generation.